kairos-contracts/demo-contract/contract/src/main.rs

```diff
@@ -81,19 +88,69 @@ pub extern "C" fn deposit() {
     casper_event_standard::emit(new_deposit_record);
 }
 
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct Proof {
+    pub receipt: risc0_zkvm::Receipt,
+    pub program_id: [u32; 8],
+}
+
+#[no_mangle]
+pub extern "C" fn submit_batch() {
+    let proof_serialized: Vec<u8> = runtime::get_named_arg(RUNTIME_ARG_RECEIPT);
+    let proof: Proof = from_slice(&proof_serialized).unwrap();
+    match proof.receipt.verify(proof.program_id) {
+        Ok(_) => {},
+        // replace ApiError with meaningful UserError
+        Err(_) => runtime::revert(ApiError::User(0u16))
+    };
+    // extract the proof outputs from the journal / deserialize the receipt journal
+    let journal: ProofOutputs = serde_json_wasm::from_slice(&proof.receipt.journal.bytes).unwrap();
```

Comment:
We need to get `ProofOutputs` from `receipt.verify` not the receipt.

Reply:
Could you explain this please? The ProofOutputs are extracted from the receipt only after it has been verified. If the proof is invalid then we never get to the point of deserialization.

Reply:
I have tested this code and it extracts the ProofOutputs from the receipt as expected, I don't think that there is something wrong here. 

1. verify proof
2. if proof is valid, extract outputs from receipt

Reply:
Ah your right I was misremembering verify returns `Ok(())`.
We should still get the outputs from [kairos-verifier-risc0-lib](https://github.com/cspr-rad/kairos/blob/main/kairos-prover/kairos-verifier-risc0-lib/src/lib.rs), which calls `receipt.journal.decode()`, which uses the more efficient risc0 serde instead of json.